python/demo/undocumented/helmholtz2D/helmholtz_plane_wave.py

```diff
@@ -0,0 +1,70 @@
+from dolfin import *
+from dolfin.fem.assembling import assemble
+from dolfin.io import XDMFFile
+import numpy as np
+
+# Test Helmholtz problem for which the exact solution is a plane wave
+# propagating at angle theta to the postive x-axis.
+# Chosen so we can compare with results from Ihlenburg's book
+# "Finite Element Analysis of Acoustic Scattering" p138-139
+
+# Parameters 
+k0 = 10        # wavenumber
+deg = 1        # degree of approximating polynomials
+n_elem = 128   # number of elements in each direction of mesh
+
+# Mesh
+mesh = UnitSquareMesh(MPI.comm_world, n_elem, n_elem)
+n = FacetNormal(mesh) 
+
+# Incident plane wave
+theta = np.pi/8
+ui = Expression(
```

Comment:
We should aim at UFL expression instead of C++ Expression whenever possible. See https://github.com/FEniCS/dolfinx/pull/207 and try to replace this.